coinSums.rb

```diff
@@ -0,0 +1,27 @@
+# The idea is to use recursion. Each recursive cycle will either remove the highest allowed denomination or reduce the amount.
+#
+# Here I use a technique called memoization to improve performance. Though recursion works fine for small amounts, memoization improves calculation speed for big amounts by remembering calculations it has done before (it stores them in a hashmap which has linear lookup time).
+
+
+class CoinSum
+  $sums = Hash.new();
```

Comment:
This code looks like it would work just fine without needing a global variable. Try removing the $ and see. Globals are generally a bad idea<sup>TM</sup>, because you then lose the ability to encapsulate your code into a single method, running the risk that you've declared `sums` somewhere else, or increase the number of invisible dependencies between your objects.